src/SWP/Component/MultiTenancy/Factory/TenantFactory.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher MultiTenancy Component.
+ *
+ * Copyright 2016 Sourcefabric z.u. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Component\MultiTenancy\Factory;
+
+/**
+ * Class TenantFactory.
+ */
+class TenantFactory implements TenantFactoryInterface
+{
+    /**
+     * @var string
+     */
+    protected $className;
+
+    /**
+     * TenantFactory constructor.
+     *
+     * @param string $className
+     */
+    public function __construct($className)
+    {
+        $this->className = $className;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create()
```

Comment:
as it can't be chained then in factory it's better to make it static. Then instead keeping it in container and fetching if from it every time we could just do:
```
use SWP\Component\MultiTenancy\Model\Tenant;
$tenant = TenantFactory::create(Tenant::class);
```
And as class name is only stored in that service thing then i don't see why it should be in container.